lib/phantomManager.js

```diff
@@ -116,9 +119,15 @@ PhantomManager.prototype._executeInWorker = function (worker, options, cb) {
     this._timeouts.push(timeout);
 
     worker.execute(options, function (err, result) {
-        if (err)
+        if (isDone)
```

Comment:
It looks to me this is the main fix. Attribute `callbackWasCalled` is just duplicating `isDone`.

```js
worker.execute(options, function (err, result) {
   if (isDone) {
        return; //timeout has took the responsibility to callback
   }

   isDone = true; //we are taking responsibility to callback
   
   if (err) { 
       cb(err);
   }
   ....
```